gobblin-core/src/main/java/gobblin/metrics/GobblinMetrics.java

```diff
@@ -118,12 +120,17 @@ public synchronized static GobblinMetrics get(String id, MetricContext parentCon
    * @param tags the given list of {@link Tag}s
    * @return a {@link GobblinMetrics} instance
    */
-  public synchronized static GobblinMetrics get(String id, MetricContext parentContext, List<Tag<?>> tags) {
-    GobblinMetricsRegistry registry = GobblinMetricsRegistry.getInstance();
-    if (!registry.containsKey(id)) {
-      registry.putIfAbsent(id, new GobblinMetrics(id, parentContext, tags));
-    }
-    return registry.get(id);
+  public static GobblinMetrics get(String id, MetricContext parentContext, List<Tag<?>> tags) {
+    return GOBBLIN_METRICS_REGISTRY.putIfAbsent(id, new GobblinMetrics(id, parentContext, tags));
```

Comment:
`putIfAbsent` returns "the previous value associated with the specified key, or null if there was no mapping for the key.". So I think you should return `GOBBLIN_METRICS_REGISTRY.get(id)`.